### PR TITLE
XLS-65/66: Add LendingProtocolV1_1 two-step loan creation

### PR DESCRIPTION
**Pull request by Tapanito**
## Summary

- **XLS-65**: Adds `AssetsReserved` to the Vault ledger entry and `vault_info` RPC response under `LendingProtocolV1_1`. This field tracks assets reserved for pending Loans that have not yet been accepted.
- **XLS-66**: Introduces a two-step loan creation flow under `LendingProtocolV1_1` that eliminates the custom dual-signature requirement:
  - Adds `Borrower` and `StartDate` fields to `LoanSet` for creating a pending Loan proposal
  - Adds a separate `LoanAccept` transaction whose required transaction-specific field is `LoanID`
  - Adds the `lsfLoanPending` flag to the `Loan` ledger entry
  - Makes `OwnerNode` conditional until acceptance
  - Documents both signing flows (single-transaction and two-step)
  - Adds failure conditions and state changes for proposal, acceptance, and pending Loan deletion
  - Adds pending Loan checks to `LoanManage` and `LoanPay`

## Motivation

The current `LoanSet` dual-signature workflow creates custody lock-in and limits adoption to custodians willing to build bespoke integrations. The two-step flow (propose + accept) uses standard XRPL transactions, enabling any wallet or custodian to participate.

## Test plan

- [ ] Review all new/modified field tables for correctness (Amendment column, types, defaults)
- [ ] Verify failure conditions cover edge cases (expired proposals, pending Loan operations)
- [ ] Confirm example JSON is valid and consistent with field definitions
- [ ] Cross-check section numbering and internal cross-references

### CODE REVIEW

File: XLS-0066-lending-protocol/README.md [hunk 1]
```diff
@@ -1037,6 +1059,22 @@ Either of the parties (Borrower or Loan Issuer) may initiate the transaction. Th
...
+
+- **Step 1 — Proposal** (Loan Broker):
+  1. The `LoanBroker.Owner` creates a `LoanSet` transaction, setting the loan terms and the `Borrower` field to the Borrower's account ID.
+  2. The `LoanBroker.Owner` signs and submits the transaction.
+  3. A `Loan` object is created in a **pending** state (`lsfLoanPending` flag is set). Funds are not transferred; instead, `Vault.AssetsAvailable` is decremented and `Vault.AssetsReserved` is incremented by `PrincipalRequested`.
+  4. The Loan must be accepted before the `StartDate`. If `Loan` is not accepted by the `StartDate`, the proposal expires.
+
+- **Step 2 — Acceptance** (Borrower):
+  1. The `Borrower` creates a `LoanSet` transaction with the `tfLoanSetAccept` flag set and the `LoanID` field pointing to the pending `Loan` object.
+  2. The `Borrower` signs and submits the transaction. No other fields are required.
```
Comment (ximinez) [inline XLS-0066-lending-protocol/README.md]:
```suggestion
  2. The `Borrower` signs and submits the transaction. No other fields are permitted.
```

File: XLS-0066-lending-protocol/README.md [hunk 2]
```diff
@@ -1195,6 +1361,21 @@ This transaction uses the standard transaction fee.
...
+1. Remove `LoanID` from the `OwnerDirectory` of the `LoanBroker` _pseudo-account_ (using `LoanBrokerNode`).
+2. Delete the `Loan` object.
+3. Update `Vault` object:
+   - Increase `Vault.AssetsAvailable` by `Loan.PrincipalOutstanding`.
+   - Decrease `Vault.AssetsReserved` by `Loan.PrincipalOutstanding`.
+   - Decrease `Vault.AssetsTotal` by `InterestDue` (reverses the interest that was added at proposal time).
+4. Update `LoanBroker` object:
+   - Decrease `LoanBroker.DebtTotal` by `Loan.PrincipalOutstanding + InterestDue`.
+   - Decrement `LoanBroker.OwnerCount` by `1`.
+5. Release the reserve from the Loan Broker: Decrement `AccountRoot(LoanBroker.Owner).OwnerCount` by `1`.
```
Comment (ximinez) [inline XLS-0066-lending-protocol/README.md:1393]: This is an inappropriate amount of work to do in the case of a `LoanSet` acceptance returning  `tecEXPIRED` (3.8.6.2). All other "post-`tec`" modifications only delete objects (and remove them from directories). This would be modifying objects, changing numeric values. It goes beyond cleanup.

I'm not sure how to square the circle, but here are some options:
* Just fail. Don't delete the loan.
* Return `tesSUCCESS` _as if_ the `LoanSet` was really a `LoanDelete`. Not a great UX if the user was expecting the Loan to succeed.
	* It might be reasonable if you create a new `LoanSet` flag. Either "delete if expired" or "_don't_ delete if expired." The existence of either flag would indicate that _not_ having it set will do the opposite, and we can define that opposite as the default behavior, and thus sensibly return `tesSUCCESS`.
* Create a new `tes` code to reflect that the transaction did something other than what was requested. `tesDELETED`. This is also not a great option, because it'll affect every integration that checks for the string result for `tesSUCCESS` instead of anything starting with `tes`. I'd bet that's most of them. You'd also have to compete with the `tec` codes to find an appropriate numeric value.

**Review by ximinez — DISMISSED** [review]

File: XLS-0066-lending-protocol/README.md (hunk 2 above)
Comment (Tapanito) [reply XLS-0066-lending-protocol/README.md:1393]: > Just fail. Don't delete the loan.

On further deliberation, yeah it makes sense to not delete the Loan implicitly, and rather rely on explicit call to `LoanDelete`.

File: XLS-0066-lending-protocol/README.md [hunk 3]
```diff
@@ -962,27 +978,28 @@ The transaction creates a new `Loan` object.
...
+| `LoanBrokerID`            |   Yes    | `string`  |   `HASH256`   |     `N/A`     |                       | The Loan Broker ID associated with the loan.                                                                                                  |
+| `Flags`                   |    No    | `number`  |   `UINT32`    |       0       |                       | Specifies the flags for the Loan.                                                                                                             |
+| `Data`                    |    No    | `string`  |    `BLOB`     |     None      |                       | Arbitrary metadata in hex format. The field is limited to 256 bytes.                                                                          |
+| `Borrower`                |    No    | `string`  |  `ACCOUNTID`  |     `N/A`     | `LendingProtocolV1_1` | The address of the Borrower. Used in the two-step loan creation flow. Only the `LoanBroker.Owner` may set this field.                         |
```
Comment (a1q123456) [inline XLS-0066-lending-protocol/README.md:987]: Seems that we also need to add `StartDate` and LoanSet is in the 2-step flow if 
1) CounterParty is not specified, and
2) CounterPartySignature is not specified, and
3) Borrower is specified, and
4) StartDate is specified

File: XLS-0066-lending-protocol/README.md [hunk 4]
```diff
@@ -1037,6 +1058,22 @@ Either of the parties (Borrower or Loan Issuer) may initiate the transaction. Th
...
 
+##### 3.8.3.2 Two-Step Flow (`LendingProtocolV1_1`)
+
+The two-step flow eliminates the need for custom multi-party signing. Instead, loan creation is split into two independent, standard-signature transactions.
+
+- **Step 1 — Proposal** (Loan Broker):
+  1. The `LoanBroker.Owner` creates a `LoanSet` transaction, setting the loan terms and the `Borrower` field to the Borrower's account ID.
+  2. The `LoanBroker.Owner` signs and submits the transaction.
+  3. A `Loan` object is created in a **pending** state (`lsfLoanPending` flag is set). Funds are not transferred; instead, `Vault.AssetsAvailable` is decremented and `Vault.AssetsReserved` is incremented by `PrincipalRequested`.
+  4. The Loan must be accepted before the `StartDate`. If `Loan` is not accepted by the `StartDate`, the proposal expires. `LoanDelete` must be submitted to reclaim `Vault.AssetsReserved`.
```
Comment (a1q123456) [inline XLS-0066-lending-protocol/README.md:1069]: Should LoanSet validate the proposed start date? What should we return if it's in the past?

Comment (Tapanito) [reply XLS-0066-lending-protocol/README.md:1069]: Yeah, it should. Good call!

File: XLS-0066-lending-protocol/README.md [hunk 5]
```diff
@@ -1046,21 +1083,23 @@ The account specified in the `Account` field pays the transaction fee.
 ##### 3.8.5.1 Data Verification
```
Comment (a1q123456) [inline XLS-0066-lending-protocol/README.md]: also past-StartDate => tecEXPIRED

File: XLS-0066-lending-protocol/README.md (hunk 2 above)
Comment (Tapanito) [reply XLS-0066-lending-protocol/README.md:1409]: Taken the first option. An expired `LoanAccept` now just fails: the specification states that the pending `Loan` is not deleted and none of its fields are modified, and that reclaiming the reserved assets requires an explicit `LoanDelete` (§3.9.3.2). No new flag, `tes` code, or post-`tec` object modification is involved.

File: XLS-0066-lending-protocol/README.md (hunk 3 above)
Comment (Tapanito) [reply XLS-0066-lending-protocol/README.md]: Added. `StartDate` is now a `LoanSet` field, and §3.8.3 states the discriminator as exactly those four conditions. `Borrower`, `StartDate`, and `CounterpartySignature` are conditionally required, with `temINVALID` when `StartDate` is combined with `Counterparty`/`CounterpartySignature` or when `Borrower` appears without `StartDate`.

File: XLS-0066-lending-protocol/README.md (hunk 4 above)
Comment (Tapanito) [reply XLS-0066-lending-protocol/README.md]: Specified. A proposal whose `StartDate` is at or before the current ledger timestamp fails with `tecEXPIRED`, so it cannot be created already expired (§3.8.5.3). The comparison is inclusive, matching how `tecEXPIRED` is evaluated elsewhere against the parent close time.

File: XLS-0066-lending-protocol/README.md [hunk 6]
```diff
@@ -1398,7 +1613,7 @@ First, the system determines the final destination of all funds.
 
 The `Loan` object is updated to reflect the payment.
 
-3. If the loan was impaired (`lsfLoanImpaired` flag was set), the loan is unimpaired before the payment is processed (see [LoanManage tfLoanUnimpair](#3105-state-changes) for details).
+3. If the loan was impaired (`lsfLoanImpaired` flag was set), the loan is unimpaired before the payment is processed (see [LoanManage tfLoanUnimpair](#31115-state-changes) for details).
```
[the remark this answers is not in the document]
Comment (Tapanito) [reply XLS-0066-lending-protocol/README.md]: Fixed. The link now points at `#3115-state-changes`, matching the `3.11.5 State Changes` heading.

**Comment by mvadari** [comment]
@ai-review --aggressive

File: XLS-0065-single-asset-vault/README.md [hunk 7]
```diff
@@ -87,26 +87,27 @@ The key of the `Vault` object is the result of [`SHA512-Half`](https://xrpl.org/
...
+| ------------------- | :------: | :------: | :----------------: | :-----------: | :-----------: | :-------------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `LedgerEntryType`   |    No    |   Yes    |      `string`      |   `UINT16`    |   `0x0084`    |                       | Ledger object type.                                                                                                                                    |
+| `LedgerIndex`       |    No    |   Yes    |      `string`      |   `UINT16`    |     `N/A`     |                       | Ledger object identifier.                                                                                                                              |
+| `Flags`             |   Yes    |   Yes    |      `string`      |   `UINT32`    |       0       |                       | Ledger object flags.                                                                                                                                   |
```
[the remark this answers is not in the document]
Comment (Tapanito) [reply XLS-0065-single-asset-vault/README.md]: Fixed the Vault table to follow ledger-entry conventions: LedgerEntryType and LedgerIndex are constant, LedgerIndex is HASH256, and Flags is a non-constant UINT32 with JSON type number.

File: XLS-0066-lending-protocol/README.md [hunk 8]
```diff
@@ -403,38 +404,38 @@ The `LoanID` is calculated as follows:
...
+| -------------------------- | :------: | :---------: | :-------: | :-----------: | :-------------------------------------------------------------------------------: | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `LedgerEntryType`          |   Yes    |     Yes     | `string`  |   `UINT16`    |                                     `0x0089`                                      | Ledger object type.                                                                                                                                                           |
+| `LedgerIndex`              |   Yes    |     Yes     | `string`  |   `UINT16`    |                                       `N/A`                                       | Ledger object identifier.                                                                                                                                                     |
```
[the remark this answers is not in the document]
Comment (Tapanito) [reply XLS-0066-lending-protocol/README.md]: Fixed. Loan.LedgerIndex now uses internal type HASH256.

File: XLS-0066-lending-protocol/README.md (hunk 1 above)
Comment (Tapanito) [reply XLS-0066-lending-protocol/README.md]: Updated. The current rippled LoanAccept schema in PR #7820 requires LoanID and defines no other transaction-specific fields, so the acceptance step now says the Borrower signs and submits LoanAccept and that no other transaction-specific fields are permitted.

File: XLS-0066-lending-protocol/README.md [hunk 9]
```diff
@@ -962,27 +978,29 @@ The transaction creates a new `Loan` object.
...
+| ------------------------- | :---------: | :-------: | :-----------: | :-----------: | :-------------------- | :-------------------------------------------------------------------------------------------------------------------------------------------- |
+| `TransactionType`         |     Yes     | `string`  |   `UINT16`    |     `80`      |                       | The transaction type.                                                                                                                         |
+| `LoanBrokerID`            |     Yes     | `string`  |   `HASH256`   |     `N/A`     |                       | The Loan Broker ID associated with the loan.                                                                                                  |
+| `Flags`                   |     No      | `number`  |   `UINT32`    |       0       |                       | Specifies the flags for the Loan.                                                                                                             |
```
[the remark this answers is not in the document]
Comment (Tapanito) [reply XLS-0066-lending-protocol/README.md:985]: Updated the PR description. The two-step flow now describes LoanSet with Borrower and StartDate followed by a separate LoanAccept with LoanID; it no longer mentions LoanID on LoanSet or tfLoanSetAccept.

File: XLS-0066-lending-protocol/README.md [hunk 10]
```diff
@@ -962,27 +978,29 @@ The transaction creates a new `Loan` object.
...
+| `ClosePaymentFee`         |     No      | `string`  |   `NUMBER`    |       0       |                       | A nominal funds amount paid to the `LoanBroker.Owner` when an early full repayment is made.                                                   |
+| `OverpaymentFee`          |     No      | `number`  |   `UINT32`    |       0       |                       | A fee charged on overpayments in 1/10th basis points. Valid values are between 0 and 100000 inclusive. (0 - 100%)                             |
+| `InterestRate`            |     No      | `number`  |   `UINT32`    |       0       |                       | Annualized interest rate of the Loan in in 1/10th basis points. Valid values are between 0 and 100000 inclusive. (0 - 100%)                   |
+| `LateInterestRate`        |     No      | `number`  |   `UINT32`    |       0       |                       | A premium added to the interest rate for late payments in in 1/10th basis points. Valid values are between 0 and 100000 inclusive. (0 - 100%) |
```
[the remark this answers is not in the document]
Comment (Tapanito) [reply XLS-0066-lending-protocol/README.md]: Fixed both duplicated “in in” phrases.

File: XLS-0066-lending-protocol/README.md [hunk 11]
```diff
@@ -1158,34 +1239,151 @@ The account specified in the `Account` field pays the transaction fee.
...
+3. The `Account` submitting the transaction is not the `Loan.Borrower`. (`tecNO_PERMISSION`)
+4. The current ledger timestamp is greater than or equal to `Loan.StartDate` (the proposal has expired). (`tecEXPIRED`)
+   - The pending `Loan` object is **not** deleted and none of its fields are modified.
+   - The `Loan Broker` or the `Borrower` must submit an explicit `LoanDelete` transaction to delete the pending `Loan` and restore the reserved assets.
+5. The Borrower does not have sufficient reserve for the `Loan` object. (`tecINSUFFICIENT_RESERVE`)
+6. The Vault _pseudo-account_ is frozen for the asset. (`tecFROZEN` for IOUs, `tecLOCKED` for MPTs)
+7. The LoanBroker _pseudo-account_ is deep frozen for the asset. (`tecFROZEN` for IOUs, `tecLOCKED` for MPTs)
+8. The Borrower is frozen for the asset. (`tecFROZEN` for IOUs, `tecLOCKED` for MPTs)
+9. The `LoanBroker.Owner` is deep frozen for the asset. (`tecFROZEN` for IOUs, `tecLOCKED` for MPTs)
+10. Cannot add asset holding for the `Vault.Asset` (e.g., MPToken or TrustLine issues). (`tecNO_PERMISSION`)
```
Comment (a1q123456) [inline XLS-0066-lending-protocol/README.md:1306]: We currently return terNO_RIPPLE for IOU and tecNO_AUTH for MPT on `develop` so maybe we should update this and also for other transactors.

**Review by a1q123456 — CHANGES_REQUESTED** [review]
